### PR TITLE
Adding Spark App Tasks for running in client, cluster and YARN mode

### LINKED ISSUE #9: Port sparkapp as s-c-t module
As a developer, I'd like to move *sparkapp* module from XD to s-c-t-m repo, so I can use it as task to build batch pipelines. 

**Acceptance:**
* *sparkapp* as task module (Spring Boot App) is available in *spring-cloud-task-modules* repo 
* maven coordinates include *sparkapp* with respective group/artifact/version

### PR DESCRIPTION
**Pull request by trisberg**
- adding common configuration sub-project for shared code
- adding test for local execution (cluster and YARN need to be part of integration tests)

Resolves #9

### CODE REVIEW

**Comment by jvalkeal** [comment]
I suppose you accidentally pushed this branch into main repo, not into your own clone? Not a biggie but need to zap that branch afterwards.

File: sparkapp-client/pom.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+				<exclusion>
+					<groupId>org.slf4j</groupId>
+					<artifactId>slf4j-log4j12</artifactId>
+				</exclusion>
+			</exclusions>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
+			<version>2.4.4</version>
```
Comment (jvalkeal) [inline sparkapp-client/pom.xml:70]: Should version come from parent pom for jackson or do we require this specific versions.

File: sparkapp-cluster/pom.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+				<exclusion>
+					<groupId>org.slf4j</groupId>
+					<artifactId>slf4j-log4j12</artifactId>
+				</exclusion>
+			</exclusions>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
+			<version>2.4.4</version>
```
Comment (jvalkeal) [inline sparkapp-cluster/pom.xml:70]: Same ^^^: Should version come from parent pom for jackson or do we require this specific versions.

Comment (trisberg) [inline sparkapp-cluster/pom.xml:70]: The build pulls in a newer version that doesn't work, so we need these explicit overrides

**Comment by trisberg** [comment]
Hmm, strange, my clone wasn't configured right, so let me close this and open a new PR

**Pull request closed without merging**